rust_src/src/category.rs

```diff
@@ -11,7 +11,7 @@ use threads::ThreadState;
 #[lisp_fn]
 fn category_table_p(arg: LispObject) -> LispObject {
     LispObject::from_bool(arg.as_char_table().map_or(false, |table| {
-        LispObject::from(table.purpose).eq(LispObject::from(Qcategory_table))
+        LispObject::from(table.purpose).eq(LispObject::from(unsafe { Qcategory_table }))
```

Comment:
You should be able to do this without turning it into a LispObject along the way.